packages/cli/src/lib/updateCheck.js

```diff
@@ -268,9 +273,11 @@ export function updateCheckMiddleware(argv) {
     return
   }
 
-  if (shouldShow()) {
+  if (shouldShow() && !isLockSet(SHOW_LOCK_IDENTIFIER)) {
```

Comment:
I don't have the full picture, so maybe these suggestions don't make a lot of sense. But I think I'd prefer it this way for more consistency:

Move the `isLockSet()` check inside `shouldShow()`, because the message should never show if the lock is set (and not timed out), right?

Have the update lock check inside `shouldCheck()`

And maybe also setting the lock could be made more consistent by setting the check lock right before calling `spawnBackgroundProcess` and then clearing it again when the background process finishes (if there is a way to know when that happens)

Reply:
Happy to move the lock checks within the `should` functions - that's better. 

I haven't changed the removal of the checking lock. I'm not sure about setting event handlers on processes that then get detached. It is probably possible but I've just kept it as is since it works fine. All the logic for the update check lives in the one function so it's not too difficult to have the lock removed before that function completes.